src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.Fluent.ResourceManager/Core/RestClient/UserAgentDelegatingHandler.cs

```diff
@@ -12,25 +12,41 @@ namespace Microsoft.Azure.Management.Fluent.Resource.Core
 {
     public class UserAgentDelegatingHandler : DelegatingHandlerBase
     {
-        private string userAgent;
-        public UserAgentDelegatingHandler() : base() { }
+        private List<string> userAgents;
+        public UserAgentDelegatingHandler() : base() {
+            this.userAgents = new List<string>();
+        }
 
         public UserAgentDelegatingHandler(HttpMessageHandler innerHandler) : base(innerHandler)
         { }
 
-        public void setUserAgent(string userAgent)
+        public void SetUserAgent(string userAgent)
         {
-            this.userAgent = userAgent;
+            this.userAgents.Clear();
+            this.AppendUserAgent(userAgent);
         }
 
-        public void appendUserAgent(string userAgent)
+        public void AppendUserAgent(string userAgent)
         {
-            this.userAgent = " " + userAgent;
+            this.userAgents.Add(userAgent);
         }
 
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            request.Headers.Add("User-Agent", this.userAgent);
+            if (this.userAgents.Any())
+            {
+                if (request.Headers.UserAgent != null)
+                {
+                    this.userAgents.All(userAgent => {
```

Comment:
Although All is doing the trick here, but it is better not to use, as All is used to check a condition, which we are not doing here. I would use a foreach instead.